src/makeInfinityAdder.js

```diff
@@ -4,9 +4,22 @@
  *
  * @return {function}
  */
+let sum = 0;
 
 function makeInfinityAdder() {
-  // write code here
+  return (number) => {
+    if (number) {
+      sum += number;
+
+      return makeInfinityAdder();
+    } else {
+      const result = sum;
+
+      sum = 0;
+
+      return result;
```

Comment:
we can return only value from parameter in **makeInfinityAdder**  without 
_const result = sum; 
sum = 0;_
because we have default value in **x** or value of numbers sum

Reply:
thanks for all your advises! I'll fix my code using your recommendations!